ПЕ-21б Поздеева О.В.: Создание репозитория комментариев для задач.

OxanaPozdeeva opened the pull request:
Создала таблицу Task.comment
Создала класс комментариев
Создала репозитории комментариев с методами
Для каждого метода был создан тест

OxanaPozdeeva committed:
ПЕ-21б Поздеева О.В.: Создание репозиторий комментариев для задач.

Создала таблицу Task.comment
Создала класс комментариев
Создала репозитории комментариев с методами
Для каждого метода был создан тест

OxanaPozdeeva committed:
ПЕ-21б Поздеева О.В.: Создание репозитория комментариев для задач.

Изменила название поля text_comment на text в таблице.
Изменила тип поля даты в таблице на datetime.
Убрала лишние строки в файле проекта IS.Model.csproj.
Добавила неиспользованные ранее поля в методах.

dha01 commented on IS.Model/IS.Model/IS.Model.csproj:
```diff
@@ -89,7 +92,23 @@
     <Compile Include="Service\UserService.cs" />
   </ItemGroup>
-  <ItemGroup />
+  <ItemGroup>
```
Ниже какие то ненужные изменения

dha01 commented on SQL/Task_27/create table Task.comment.sql:
```diff
@@ -0,0 +1,15 @@
+	comment int identity,
+	add_date date,
+	person int,
+	text_comment nvarchar(max) not null,
```
просто text это и так таблица комментариев

dha01 commented on IS.Model/IS.Model.Tests/Repository/Comment/CommentRepositoryTests.cs:
```diff
@@ -0,0 +1,166 @@
+			_comment = new CommentItem()
+			{
+				AddDate = DateTime.Now.Date, 
+				Person = 2,
```
Нужно использовать репозитории персональных данных и задач для того чтоб в тесте создавались записи для тестирования и тут присвоить идентификаторы этих созданных записей.
т.к. в базе может не существовать записей с такими идентификаторами и тесты не пройдут.

dha01 commented on IS.Model/IS.Model.Tests/Repository/Comment/CommentRepositoryTests.cs:
```diff
@@ -0,0 +1,166 @@
+		/// <summary>
+		/// Проверяет еквивалентны ли два комментария.
+		/// </summary>
+		/// <param name="first_comment"></param>
```
Добавить описание входных параметров.

dha01 commented on IS.Model/IS.Model/IS.Model.csproj:
```diff
@@ -89,7 +92,6 @@
     <Compile Include="Service\TaskService.cs" />
     <Compile Include="Service\UserService.cs" />
   </ItemGroup>
-  <ItemGroup />
```
Удалена лишняя строка.

dha01 commented on IS.Model/IS.Model/Item/Comment/CommentItem.cs:
```diff
@@ -0,0 +1,38 @@
+		/// <summary>
+		/// Автор.
+		/// </summary>
+		public int Person { get; set; }
```
Чтобы отличать данные от идентификаторов такие поля в С# следует именовать как PersonId и TaskId

dha01 commented on IS.Model/IS.Model/IS.Model.csproj:
```diff
@@ -55,6 +56,8 @@
     <Compile Include="Properties\AssemblyInfo.cs" />
     <Compile Include="Repository\Access\IRoleRepository.cs" />
+    <Compile Include="Repository\Comment\CommentRepository.cs" />
+    <Compile Include="Repository\Comment\ICommentRepository.cs" />
```
Ссылка есть, но сам файл потерялся.

OxanaPozdeeva committed:
ПЕ-21б Поздеева О.В.: Создание репозитория комментариев для задач.

Вернула потерянный файл
Переименовала поля PersonId и TaskId
Вернула удаленную строку в файл проекта IS.Model.csproj
Добавила описание входных параметров
Создала в тесте записи для тестирования

OxanaPozdeeva committed:
ПЕ-21б Поздеева О.В.: Создание репозитория комментариев

Убрала лишние строки в CommentItem.cs
